src/Unirest/Request.php

```diff
@@ -794,47 +794,32 @@ public static function getFormattedHeaders($headers)
         return $formattedHeaders;
     }
 
-    private static function getArrayFromQuerystring($query)
-    {
-        $query = preg_replace_callback('/(?:^|(?<=&))[^=[]+/', function ($match) {
-            return bin2hex(urldecode($match[0]));
-        }, $query);
-
-        parse_str($query, $values);
-
-        return array_combine(array_map('hex2bin', array_keys($values)), $values);
-    }
-
     /**
-     * Ensure that a URL is encoded and safe to use with cURL
-     * @param  string $url URL to encode
-     * @return string
+     * Validates and processes the given Url to ensure safe usage with cURL.
+     * @param string $url The given Url to process
+     * @return string Pre-processed Url as string
+     * @throws Exception
      */
-    private static function encodeUrl($url)
+    public static function cleanUrl($url)
```

Comment:
cleanUrl is not what the function is doing rather it should be something validateAndCleanUrl or any other better name. As this method is doing two things i.e. validation of url and cleaning url in case of double slashes.

Reply:
since removing slashes are also same thing as validating the url hence renaming it to `validateUrl`

Reply:
since removing slashes are also same thing as validating the url hence renaming it to `validateUrl`